src/components/calcite-radio-button/calcite-radio-button.tsx

```diff
@@ -0,0 +1,251 @@
+import {
+  Component,
+  Host,
+  h,
+  Listen,
+  Prop,
+  Element,
+  Watch,
+  Event,
+  EventEmitter,
+} from "@stencil/core";
+import { guid } from "../../utils/guid";
+
+@Component({
+  tag: "calcite-radio-button",
+  styleUrl: "calcite-radio-button.scss",
+  shadow: true,
+})
+export class CalciteRadioButton {
+  constructor() {
+    this.check = this.check.bind(this);
```

Comment:
This isn't necessary. I was wrong, event listeners **are** bound to the instance, so you can convert fields holding arrow functions (`onInputBlur`, `checkFirstRadioButton`, and `uncheckOtherRadioButtonsInGroup`) to class methods and just use `bind` in  `renderHiddenRadioInput`:

```ts
  private renderHiddenRadioInput() {
    // ...
    this.input.onfocus = this.check.bind(this);
    this.input.onblur = this.onInputBlur.bind(this);
    // ...
```

Reply:
Weird, I'm not seeing that behavior.  When I remove `constructor() { this.check = this.check.bind(this); }` I see this on the console when I click a radio button:
```
calcite-radio-button.entry.js:100 Uncaught TypeError: this.uncheckOtherRadioButtonsInGroup is not a function
    at HTMLInputElement.check (calcite-radio-button.entry.js:100)
    at CalciteRadioButton.onFocusedChange (calcite-radio-button.entry.js:69)
    at index-b00ef87e.js:1810
    at Array.map (<anonymous>)
    at setValue (index-b00ef87e.js:1807)
    at CalciteRadioButton.set [as focused] (index-b00ef87e.js:1860)
    at CalciteRadioButton.check (calcite-radio-button.entry.js:101)
    at HTMLElement.<anonymous> (index-b00ef87e.js:87)
```
`this.click()` is called in two different places, both when the hidden radio is focused with the keyboard and when it is clicked with the mouse, and for some reason even though it has the `@Listen` decorator, it doesn't seem to be binding `this` correctly.

I was able to remove the arrow functions on `onInputBlur`, `checkFirstRadioButton` and `uncheckOtherRadioButtonsInGroup` though.

Reply:
`this.check.bind(this)` does work.  I'll include that in the next update.